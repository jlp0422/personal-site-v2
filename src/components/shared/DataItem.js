import styled from '@emotion/styled'
import React from 'react'

const SIZES = {
  small: {
    value: '1.6rem',
    label: '1rem'
  },
  medium: {
    value: '2.0rem',
    label: '1.4rem'
  },
  large: {
    value: '2.6rem',
    label: '1.8rem'
  }
}

const alignToFlex = {
  center: 'center',
  left: 'flex-start',
  right: 'flex-end'
}

const isDark = background => background === 'dark'

export const DataWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: ${({ align }) => alignToFlex[align]};
  margin: 0;
  > p {
    margin: 0;
  }
  > p:first-of-type {
    font-size: ${({ size }) => SIZES[size].value};
    color: ${({ theme, background }) =>
      isDark(background)
        ? theme.palette.white.lightest
        : theme.palette.text.primary};
  }
  > p:last-of-type {
    text-transform: uppercase;
    font-weight: 300;
    font-size: ${({ size }) => SIZES[size].label};
    color: ${({ theme, background }) =>
      isDark(background)
        ? theme.palette.white.lighter
        : theme.palette.text.gray};
  }
`

const DataItem = ({ value, label, size, align, background }) => {
  return (
    <DataWrapper size={size} align={align} background={background}>
      <p>{value}</p>
      <p>{label}</p>
    </DataWrapper>
  )
}

export default DataItem
