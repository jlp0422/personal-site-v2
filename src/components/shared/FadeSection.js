import React, { useState, useRef, useEffect } from 'react'
import styled from '@emotion/styled'

const Container = styled.div`
  display: inherit;
  opacity: ${({ isVisible }) => (isVisible ? 1 : 0)};
  transform: ${({ isVisible }) => (isVisible ? 'none' : 'translateY(5vh)')};
  visibility: ${({ isVisible }) => (isVisible ? 'visible' : 'hidden')};
  transition: opacity 0.5s ease-out, transform 0.5s ease-out;
  will-change: opacity, visibility;
  margin-top: 0;
`

const FadeSection = ({ children }) => {
  const [isVisible, setVisible] = useState(false)
  const domRef = useRef()
  useEffect(() => {
    const observer = new IntersectionObserver((entries, o) => {
      entries.forEach(entry => {
        if (entry.isIntersecting) {
          setVisible(true)
        }
      })
    })
    observer.observe(domRef.current)
    return () => {
      observer.unobserve(domRef.current)
    }
  }, [])

  return (
    <Container isVisible={isVisible} ref={domRef}>
      {children}
    </Container>
  )
}

export default FadeSection
