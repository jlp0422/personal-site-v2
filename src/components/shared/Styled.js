import styled from '@emotion/styled'
import { useTheme } from 'emotion-theming'
import { Link } from 'gatsby'
import { shade } from 'polished'
import React from 'react'

export const OutboundLink = styled.a`
  text-decoration: none;
  margin: 0 0 0 0.6rem;
  font-weight: bold;
  color: ${({ theme }) => theme.palette.link.secondary};
  --initialBgColor: ${({ theme }) => theme.palette.link.secondary};/* 000; */
  --slideBgColor: #fff;

  background-image: linear-gradient(
    90deg,
    var(--slideBgColor) 0%,
    var(--slideBgColor) 50%,
    var(--initialBgColor) 50%,
    var(--initialBgColor) 100%
  );
  background-size: 200%;
  transition: background-position 0.2s ease-out,
    color 0.2s ease-out;
  transition-delay: 0s, 0.15s;
  :hover {
    color: #fff;
    background-position: -100% 100%;
  }
`

export const GridWrapper = styled.div`
  display: grid;
  grid-template-columns: 2rem 1fr;
  align-items: center;
  justify-items: start;
  padding: 0.5rem 0;
  > a {
    padding: 0.4rem;
  }
`

export const NavLink = styled(Link)`
  color: ${({ theme }) => theme.palette.link.primary};
  margin: 0;
  padding: 1rem;
  font-family: ${({ theme }) => theme.fonts.nav};
  font-size: 2rem;
  font-weight: 700;
  text-decoration: none;
  text-transform: uppercase;
  :hover {
    transform: scale(1.05);
    color: ${({ theme }) => shade(0.2, theme.palette.link.primary)};
  }
  :active {
    transform: scale(0.95);
  }
`

export const ActiveLink = ({ children, ...props }) => {
  const { palette } = useTheme()
  return (
    <NavLink
      {...props}
      activeStyle={{
        textDecoration: 'line-through',
        textDecorationColor: palette.white.true
      }}
    >
      {children}
    </NavLink>
  )
}

export const ExpContainer = styled.div`
  background-color: ${({ theme }) => theme.palette.black.darkest};
  display: grid;
  text-align: center;
  padding: 1.5rem;
  margin: 2.5rem auto;
  border: 1px solid black;
  transition: all 0.2s linear;
  --webkit-transition: all 0.2s linear;
  :first-of-type {
    margin-top: 1rem;
  }
  :hover {
    box-shadow: ${({ theme }) => theme.palette.boxShadow.primary};
    transform: translate(-8px, -8px);
    --webkit-transform: translate(-8px, -8px);
  }
  @media only screen and (max-width: 640px) {
    box-shadow: ${({ theme }) => theme.palette.boxShadow.primary};
    :hover {
      transform: none;
    }
  }
`

export const ExpHeader = styled.h1`
  + div {
    margin-top: 1rem;
  }
`

export const ExpTitle = styled.h3`
  font-family: ${({ theme }) => theme.fonts.nav};
  color: ${({ theme }) => theme.palette.white.lightest};
`

export const ExpInfo = styled.div`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  margin: 1rem 0 0;
  @media only screen and (max-width: 640px) {
    > div {
      margin: 0.6rem 0 0;
    }
  }
`
