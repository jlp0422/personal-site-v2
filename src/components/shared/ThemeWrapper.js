import React from 'react'
import { ThemeProvider } from 'emotion-theming'
import { theme } from '../../helpers'

const ThemeWrapper = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export default ThemeWrapper
