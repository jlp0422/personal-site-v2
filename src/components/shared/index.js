export { default as DataItem } from './DataItem'
export { default as FadeSection } from './FadeSection'
export { default as GithubLink } from './GithubLink'
export { default as LiveLink } from './LiveLink'
export * from './Styled'
