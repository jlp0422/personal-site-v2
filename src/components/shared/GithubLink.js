import React from 'react'
import Image from '../Image'
import { GridWrapper, OutboundLink } from './Styled'

const Link = ({ url, linkText }) => {
  return (
    <GridWrapper>
      <Image queryKey="octocat" imgStyle={{ height: 20, width: 20 }} />
      <div>
        <OutboundLink target="_blank" href={url}>
          {linkText}
        </OutboundLink>
      </div>
    </GridWrapper>
  )
}

export default Link
