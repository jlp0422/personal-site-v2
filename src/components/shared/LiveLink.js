import React from 'react'
import Image from '../Image'
import { GridWrapper, OutboundLink } from './Styled'

const Link = ({ url, linkText }) => {
  return (
    <GridWrapper>
      <Image queryKey="computer" style={{ width: '2rem' }} imgStyle={{ margin: 0 }} />
      <OutboundLink href={url}>{linkText}</OutboundLink>
    </GridWrapper>
  )
}

export default Link
