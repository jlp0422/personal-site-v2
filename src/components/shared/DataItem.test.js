import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeWrapper } from '../shared'
import DataItem, { DataWrapper } from './DataItem'

describe('DataItem', () => {
  const getOutput = args => {
    const defaultProps = {
      value: 'Value of the prop',
      label: 'Label of the prop',
      size: 'medium',
      align: 'center',
      background: 'dark'
    }

    const props = { ...defaultProps, ...args }

    return renderer.create(
      <ThemeWrapper>
        <DataItem {...props} />
      </ThemeWrapper>
    )
  }

  it('Renders correctly', () => {
    const tree = getOutput().toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Correctly passes the props to the DataWrapper component', () => {
    const instance = getOutput().root
    const element = instance.findByType(DataWrapper)
    const getProp = key => element.props[key]
    expect(element.props.children).toHaveLength(2)
    expect(getProp('size')).toBe('medium')
    expect(getProp('align')).toBe('center')
    expect(getProp('background')).toBe('dark')
  })
})
