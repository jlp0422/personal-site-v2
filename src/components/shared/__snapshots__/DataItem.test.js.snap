// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataItem Renders correctly 1`] = `
<div
  className="css-gmm171"
  size="medium"
>
  <p>
    Value of the prop
  </p>
  <p>
    Label of the prop
  </p>
</div>
`;
